workflow/controller/operator_test.go

```diff
@@ -8512,13 +8517,18 @@ func TestMutexWfPendingWithNoPod(t *testing.T) {
 	ctx := context.Background()
 	controller.syncManager = sync.NewLockManager(GetSyncLimitFunc(ctx, controller.kubeclientset), func(key string) {
 	}, workflowExistenceFunc)
-	_, _, _, _, err := controller.syncManager.TryAcquire(wf, "test", &wfv1.Synchronization{Mutex: &wfv1.Mutex{Name: "welcome"}})
+	lockAcquired, _, msg, _, err := controller.syncManager.TryAcquire(wf, "test", &wfv1.Synchronization{Mutex: &wfv1.Mutex{Name: "welcome"}})
 	require.NoError(t, err)
 	woc := newWorkflowOperationCtx(wf, controller)
+	if !lockAcquired {
```

Comment:
This feels wrong for a test.

Surely the outcome of `TryAquire` should always be the same for this test, and therefore we could assert that `!lockAcquired`? If lockAquired is true here we're not testing the new functionality.

Reply:
done~, I modified the test case and handed over the modification of msg and status to `operate`

Reply:
This part looks good.